server/config/config.go

```diff
@@ -3,19 +3,66 @@ package config
 import (
 	"flag"
 
+	"github.com/buildbuddy-io/buildbuddy/server/util/file"
+	"github.com/buildbuddy-io/buildbuddy/server/util/log"
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+
 	flagyaml "github.com/buildbuddy-io/buildbuddy/server/util/flagutil/yaml"
 )
 
 const pathFlagName = "config_file"
 
-var configPath = flag.String(pathFlagName, "/config.yaml", "The path to a buildbuddy config file")
+var (
+	configPath = flag.String(pathFlagName, "/config.yaml", "The path to a buildbuddy config file")
+
+	// If you wish to extract a section of the config to the Config struct as opposed to parsing into flags:
+	// 1. Set the section name in configKeysToExtract so the flag parser will know to ignore it
+	// 2. Define the section in the Config struct using `yaml` tags
+	configKeysToExtract []string
```

Comment:
Example of how this would be used with migration_cache: https://github.com/buildbuddy-io/buildbuddy/pull/2486/files

Reply:
@tempoz might also have some thoughts, but would it be possible to instead add support [here](https://cs.github.com/buildbuddy-io/buildbuddy/blob/82a217bf5f5c64788ac57a038b290aba79733c0e/server/util/flagutil/types/autoflags/autoflags.go?q=path%3Aautoflags+repo%3Abuildbuddy-io%2Fbuildbuddy#L48-L52) for parsing arbitrary struct types? (Currently I think we only support struct slice types.) That way you can write something like

```
var migrationConfig = flagutil.New("cache.migration", (*MigrationConfig)(nil), "Cache migration config")
```

And then using reflection maybe it's possible to get that type `MigrationConfig` from the nil ptr, and parse using that struct type?

Reply:
woops, I see she already made the same comment :)

Reply:
👍 https://github.com/buildbuddy-io/buildbuddy/pull/2487?no-redirect=1

Reply:
As a note though, Zoey taught me that you can't have a nil pointer as the default value for a flag

"flag definitions return pointers to default-initialized data that may be set by flag.Parse (and now also the yaml population code), and nil pointers do not point to data that can be set, so the flag could never be set if it defaulted to nil"